### DISCUSSION TITLE
Retrieving composite resources via search

### DISCUSSION THREAD

**Discussion by rkodev** [opensrp/fhircore#358] [Ideas]
The search API is designed to only return results of a single resource type, i.e the search can only return a list of `Patients` or a list of `Tasks` but not both at the same.

Currently, for the Patient register in the Covax app, we need to use both the `Patient` and `Task` resources to display the patient list. One of the ways to do this in the app would be 

**a) Strategy 1**: Execute 2 queries before displaying results on the screen

-  Fetch all tasks that indicate the list of patients to be listed
-  Extract patient ids from the tasks
-  Fetch all `Patient` resources by the given ids

**b) Strategy 2** Lazy loading data in the Patient Register
- Execute a single patient search query
- Execute multiple background queries to list tasks status displayed on the view holder

We can also have a discussion on whether the `SDK` will be able to search and return composite `Resources` in the same query

** Edited to replace the word status with tasks

**Comment by fredhersch** [comment]
@rkodev Would another way to approach this be to:
1. Get a list of all the patients in the patient register (how is this defined)
2. For each patient, generate a second search query to return all tasks associated with that patient

**Comment by vincent-karuri** [comment]
@rkodev this discussion may be of interest https://github.com/opensrp/fhircore/issues/81
